src/software/backend/input/network/networking/network_filter.cpp

```diff
@@ -150,7 +151,9 @@ Ball NetworkFilter::getFilteredBallData(const std::vector<SSL_DetectionFrame> &d
         ball_filter.getFilteredData(ball_detections, field_state);
     if (new_ball_state)
     {
-        ball_state = *new_ball_state;
+        Ball new_ball = *new_ball_state;
```

Comment:
Is there a reason to assign `*new_ball_state` to `new_ball` instead of directly assigning `*new_ball_state` states to `ball_state`? Also, it might be good to update `new_ball_state` name to reflect that it's a `Ball` not `BallState`

Reply:
Good point! Forgot I had a method for that haha